File: Server/ServiceLayer/Objects/Operations/Client/RegisterOperation.java
package ServiceLayer.Objects.Operations.Client;

import ServiceLayer.Objects.ClientOperation;
import ServiceLayer.Protocol;

public class RegisterOperation extends ClientOperation {
    private String email;
    private String password;
    private String firstName;
    private String lastName;
    private String idNumber;
    private String phoneNumber;
    private String birthday;

    public RegisterOperation(short opCode) {
        super(opCode);
        this.email= "";
        this.password = "";
        this.firstName = "";
        this.lastName = "";
        this.idNumber = "";
        this.phoneNumber = "";
        this.birthday = "";
    }

    @Override
    public boolean pushByte(byte nextByte) {
        if(nextByte == ';')
            return true;
        if(nextByte==(byte)0)
        {
            if(email.equals("")){
                this.email = bytesToString();
            }
            else if(password.equals("")){
                this.password = bytesToString();
            }
            else if(firstName.equals("")){
                this.firstName = bytesToString();
            }
            else if(lastName.equals("")){
                this.lastName = bytesToString();
            }
            else if(idNumber.equals("")){
                this.idNumber = bytesToString();
            }
            else if(phoneNumber.equals("")){
                this.phoneNumber = bytesToString();
            }
            else if(birthday.equals("")){
                this.birthday = bytesToString();
            }
        }
        else
            pushNextByte(nextByte);
        return false;
    }

    public String getEmail() {
        return email;
    }

    public String getPassword() {
        return password;
    }

    public String getBirthday() {
        return birthday;
    }
    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public String getIdNumber() {
        return idNumber;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    @Override
    public void execute(Protocol protocol) {
        protocol.register(this);
    }
}

File: Windows Client/FacultyManager/FacultyManager/Model/Operations/ClientOperations/RegisterOperation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FacultyManager.Model.Operations.ClientOperations
{
    public class RegisterOperation : ClientOperation
    {
        private readonly string _email;
        private readonly string _password;
        private readonly string _firstName;
        private readonly string _lastName;
        private readonly string _idNumber;
        private readonly string _phoneNumber;
        private readonly string _birthday;


        public RegisterOperation(short opCode, string email, string password, string firstName, string lastName, string idNumber, string phoneNumber, string birthday) : base(opCode)
        {
            _email = email;
            _password = password;
            _firstName = firstName;
            _lastName = lastName;
            _idNumber = idNumber;
            _phoneNumber = phoneNumber;
            _birthday = birthday;
        }

        public override byte[] encode()
        {
            byte[] bytes = new byte[_email.Length + _password.Length + _firstName.Length + _lastName.Length + _idNumber.Length + _phoneNumber.Length + _birthday.Length + 10];
            bytes[0] = (byte)((getOpCode() >> 8) & 0xFF);
            bytes[1] = (byte)(getOpCode() & 0xFF);
            int index = 2;
            AddStringToByteArray(_email, bytes, ref index);
            bytes[index++] = 0;
            AddStringToByteArray(_password, bytes, ref index);
            bytes[index++] = 0;
            AddStringToByteArray(_firstName, bytes, ref index);
            bytes[index++] = 0;
            AddStringToByteArray(_lastName, bytes, ref index);
            bytes[index++] = 0;
            AddStringToByteArray(_idNumber, bytes, ref index);
            bytes[index++] = 0;
            AddStringToByteArray(_phoneNumber, bytes, ref index);
            bytes[index++] = 0;
            AddStringToByteArray(_birthday, bytes, ref index);
            bytes[index++] = 0;
            bytes[index] = (byte)';';
            return bytes;
        }
    }
}
